test/core/security/grpc_tls_certificate_provider_test.cc

```diff
@@ -495,8 +495,54 @@ TEST_F(GrpcTlsCertificateProviderTest,
   CancelWatch(watcher_state_1);
 }
 
-}  // namespace testing
+TEST_F(GrpcTlsCertificateProviderTest, EmptyPrivateKeyString) {
+  std::string empty_string;
+  absl::Status match_status =
+      PrivateKeyPublicKeyMatches(empty_string, cert_chain_);
+  EXPECT_EQ(match_status.code(), absl::StatusCode::kInvalidArgument);
+  EXPECT_EQ(match_status.message(), "Private key string is empty!");
+}
+
+TEST_F(GrpcTlsCertificateProviderTest, EmptyCertificateString) {
+  std::string empty_string;
+  absl::Status match_status =
+      PrivateKeyPublicKeyMatches(private_key_2_, empty_string);
+  EXPECT_EQ(match_status.code(), absl::StatusCode::kInvalidArgument);
+  EXPECT_EQ(match_status.message(), "Certificate string is empty!");
+}
 
+TEST_F(GrpcTlsCertificateProviderTest, FailedStringToX509Conversion) {
+  std::string rubbish_string = "rubbish string";
```

Comment:
Even for a random string, just directly pass "invalid_certificate"/"invalid_private_key" to matches function.

Reply:
"invalid_private_key"